src/platform/monitoring/frontend-metrics/tests/metrics.unit.spec.js

```diff
@@ -0,0 +1,88 @@
+import * as metrics from '../metrics';
+import { navEntry } from './exampleNavigationTimingEntry';
+
+describe('metrics', () => {
+  beforeEach(() => {
+    function FormDataMock() {
+      this.append = sinon.spy();
+      this.get = sinon.mock();
+    }
+    global.FormData = FormDataMock;
+
+    const byTypeStub = sinon.stub()
+      .withArgs('navigation').returns([navEntry]);
+
+    function PerformanceMock() {
+      this.getEntriesByType = byTypeStub;
+      this.getEntriesByName = sinon.stub();
+    }
+    global.performance = new PerformanceMock();
+
+    global.window = { location: { pathname: '/' } };
+  });
+
+  describe('buildMetricsPayload', () => {
+    function buildAndParsePayload(entry) {
+      const metricsPayload = metrics.buildMetricsPayload(entry);
+      return JSON.parse(metricsPayload.append.args[0][1]);
+    }
+
+    it('should return a FormData object', () => {
+      const metricsPayload = metrics.buildMetricsPayload(navEntry);
+      expect(metricsPayload).to.be.instanceof(FormData);
+    });
+
+    it('should not include firstContentfulPaint if not present', () => {
+      const byNameStub = sinon.stub()
+        .withArgs('first-contentful-paint').returns('0');
+      global.performance.getEntriesByName = byNameStub;
+
+      const parsedResponse = buildAndParsePayload(navEntry);
+      expect(parsedResponse.metrics).to.not.include
+        .deep.members([{ duration: 1349.599999939381, metric: 'firstContentfulPaint' }]);
+    });
+
+    it('should include firstContentfulPaint if present', () => {
+      const byNameStub = sinon.stub()
+        .withArgs('first-contentful-paint').returns([{
+          duration: 0,
+          entryType: 'paint',
+          name: 'first-contentful-paint',
+          startTime: 1349.599999939381
+        }]);
+      global.performance.getEntriesByName = byNameStub;
+
+      const parsedResponse = buildAndParsePayload(navEntry);
+      expect(parsedResponse.metrics).to.include
+        .deep.members([{ duration: 1349.599999939381, metric: 'firstContentfulPaint' }]);
+    });
+
+    describe('sendMetricsToBackend', () => {
+      it('should call sendBeacon once', () => {
+        const spy = sinon.spy();
+        function NavigatorMock() {
+          this.sendBeacon = spy; // sinon.stub().returns(true);
+        }
+        global.navigator = new NavigatorMock();
+
+        const metricsPayload = metrics.buildMetricsPayload(navEntry);
+        metrics.sendMetricsToBackend(metricsPayload);
+
+        expect(spy.callCount).to.be.equal(1);
+      });
+    });
+
+    describe('captureMetrics', () => {
+      xit('should instantiate a new PerformanceObserver', () => {
+        const mock = sinon.mock(PerformanceObserver());
+        window.PerformanceObserver = sinon.stub(PerformanceObserver.prototype, 'constructor').returns(mock);
+        // TODO: This fails, in captureMetrics() with PerformanceObserver is not defined.
+        // How can we test the constructor is called here?
```

Comment:
@kfrz will this work?

```

\\ PerformanceMock constructor is called in beforeEach: global.performance = new PerformanceMock();

const performanceObserverFunc = sinon.spy(window, 'PerformanceObserver');
expect(performanceObserverFunc.called).to.be.equal(true);
```

Reply:
@jbalboni  Yes! Updated to use `global.PerformanceObserver = sinon.spy();` and it passes as expected. Thanks!